aries_vcx/src/new_protocols/mod.rs

```diff
@@ -0,0 +1,425 @@
+        }
+
+        fn anoncreds(&self) -> &Self::Anoncreds {
+            &self.anoncreds
+        }
+    }
+
+    async fn indy_teardown(pool_handle: PoolHandle, pool_name: String) {
+        indy_close_pool(pool_handle).await.unwrap();
+        indy_delete_pool(&pool_name).await.unwrap();
+    }
+
+    async fn build_msg_handler(pool_handle: i32) -> (String, MsgHandler) {
+        let (did, wallet_handle) = setup_issuer_wallet().await;
+
+        let wallet = IndySdkWallet::new(wallet_handle);
+        let anoncreds = IndySdkAnonCreds::new(wallet_handle);
+        let ledger_read = IndySdkLedgerRead::new(wallet_handle, pool_handle);
+        let ledger_write = IndySdkLedgerWrite::new(wallet_handle, pool_handle);
+
+        anoncreds
+            .prover_create_link_secret(settings::DEFAULT_LINK_SECRET_ALIAS)
+            .await
+            .unwrap();
+
+        let msg_handler = MsgHandler {
+            ledger_read,
+            ledger_write,
+            wallet,
+            anoncreds,
+        };
+
+        (did, msg_handler)
+    }
+
+    async fn _test_connection_handler(pool_handle: i32) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let (_, faber) = build_msg_handler(pool_handle).await;
+        let (_, alice) = build_msg_handler(pool_handle).await;
+
+        let dummy_endpoint = Url::parse("https://dummy.dummy/dummy")?;
+        let (_, faber_invite_verkey) = faber.wallet().create_and_store_my_did(None, None).await?;
+
+        let invitation_content = PairwiseInvitationContent::new(
+            "faber".to_owned(),
+            vec![faber_invite_verkey.clone()],
+            vec![],
+            dummy_endpoint.clone(),
+        );
+        let invitation = Invitation::with_decorators(
+            Uuid::new_v4().to_string(),
+            InvitationContent::Pairwise(invitation_content),
+            Default::default(),
+        );
+
+        let request_id = Uuid::new_v4().to_string();
+
+        let thread = match &invitation.content {
+            InvitationContent::Public(_) => {
+                let mut thread = Thread::new(request_id.clone());
+                thread.pthid = Some(invitation.id.clone());
+                thread
+            }
+            InvitationContent::Pairwise(_) => Thread::new(invitation.id.clone()),
+            InvitationContent::PairwiseDID(_) => Thread::new(invitation.id.clone()),
+        };
+
+        // Extract info from invitation
+        let bootstrap_info = alice.process_connection_invitation(invitation.content).await?;
+
+        // Build request
+        let (alice_did, alice_verkey) = alice.wallet().create_and_store_my_did(None, None).await?;
+        let mut did_doc = AriesDidDoc {
+            id: alice_did.clone(),
+            ..Default::default()
+        };
+        did_doc.set_service_endpoint(bootstrap_info.service_endpoint);
+        did_doc.set_routing_keys(bootstrap_info.routing_keys);
+        did_doc.set_recipient_keys(bootstrap_info.recipient_keys);
```

Comment:
this is essentially transformation of invitation into diddoc, could be a function ( i believe something like that exists in the codebase already, thought not sure if it's reusable here)

Reply:
It's more than that.

I added the `BootstrapInfo` in the view of also maintaining those `DID` info (like `service_endpoint_did`) in case you might wanna re-read it from the ledger. 

No idea if there are use cases for that and, if there aren't, we can simply resolve it to a DidDoc directly and call it a day.